pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -1035,3 +1040,30 @@ func isRolloutCompleted(crp *fleetv1beta1.ClusterResourcePlacement) bool {
 func isCRPScheduled(crp *fleetv1beta1.ClusterResourcePlacement) bool {
 	return condition.IsConditionStatusTrue(crp.GetCondition(string(fleetv1beta1.ClusterResourcePlacementScheduledConditionType)), crp.Generation)
 }
+
+func emitPlacementStatusMetric(crp *fleetv1beta1.ClusterResourcePlacement) {
+	metrics.FleetPlacementStatus.DeletePartialMatch(prometheus.Labels{"name": crp.Name})
+	// Check CRP Scheduled condition
+	status := "nil"
+	cond := crp.GetCondition(string(fleetv1beta1.ClusterResourcePlacementScheduledConditionType))
+	if !condition.IsConditionStatusTrue(cond, crp.Generation) {
+		if cond != nil {
+			status = string(cond.Status)
+		}
+		metrics.FleetPlacementStatus.WithLabelValues(crp.Name, strconv.FormatInt(crp.Generation, 10), string(fleetv1beta1.ClusterResourcePlacementScheduledConditionType), status).SetToCurrentTime()
+		return
+	}
+
+	// Check CRP expected conditions
+	expectedCondTypes := determineExpectedCRPAndResourcePlacementStatusCondType(crp)
+	for _, condType := range expectedCondTypes {
+		cond := crp.GetCondition(string(condType.ClusterResourcePlacementConditionType()))
```

Comment:
```suggestion
		cond = crp.GetCondition(string(condType.ClusterResourcePlacementConditionType()))
```
duplicate definition